var/spack/repos/builtin/packages/glew/package.py

```diff
@@ -40,6 +40,9 @@ class Glew(CMakePackage):
     conflicts("^egl", when="gl=osmesa")
     conflicts("^egl", when="gl=other")
 
+    # https://github.com/spack/spack/issues/42839
+    depends_on("glu")
+
```

Comment:
I think maybe it would be better to add

`self.define(GLEW_NO_GLU))`

If we do want to add GLU features, I think it deserves a variant and not ignoring the `OPENGL_glu_LIBRARY`

Reply:
I just found this on master, it seems they are explicitly disabling GLU for the CMake build

https://github.com/nigels-com/glew/blob/master/build/cmake/CMakeLists.txt#L61

Reply:
Finally catching up on the issue comments. I think we can probably achieve the same thing by adding the definition to the CFLAGS/CXXFLAGS for dependent projects.

Reply:
@kwryankrattiger Yes, that may make sense! Can you test and submit the changes for it?